spring-ai-core/src/main/java/org/springframework/ai/model/function/FunctionCallbackResolver.java

```diff
@@ -0,0 +1,37 @@
+/*
+* Copyright 2024 - 2024 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.model.function;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+
+/**
+ * Strategy interface for resolving {@link FunctionCallback} instances as bean from the
```

Comment:
I believe there could be an implementation for the `FunctionCallbackResolver` that doesn't depend on the application context, right?

Reply:
I can not think of any not app context case?